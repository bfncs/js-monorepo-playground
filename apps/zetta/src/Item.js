import React from 'react';
import PropTypes from 'prop-types';

const Item = ({ title, done, setDone }) => (
  <div>
    <input type="checkbox" onClick={() => setDone(!done)} />
    <span>{ title }</span>
  </div>
);

Item.propTypes = {
  title: PropTypes.string.isRequired,
  done: PropTypes.bool.isRequired,
  setDone: PropTypes.func.isRequired,
};

export default Item;