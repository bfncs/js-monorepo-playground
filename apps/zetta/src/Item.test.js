import React from 'react';
import renderer from 'react-test-renderer';
import Item from './Item';

const NO_OP = () => {};

describe('Item snapshot tests', () => {

  it('matches snapshot when done', () => {
    expect(renderer.create(
      <Item title={'title'} done={true} setDone={NO_OP}/>
    ).toJSON()).toMatchSnapshot();
  });

  it('matches snapshot when not done', () => {
    expect(renderer.create(
      <Item title={'title'} done={false} setDone={NO_OP}/>
    ).toJSON()).toMatchSnapshot();
  });

});