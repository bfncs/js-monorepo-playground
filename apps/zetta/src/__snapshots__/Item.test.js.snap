// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item snapshot tests matches snapshot when done 1`] = `
<div>
  <input
    checked={true}
    onClick={[Function]}
    type="checkbox"
  />
  <span>
    title
  </span>
</div>
`;

exports[`Item snapshot tests matches snapshot when not done 1`] = `
<div>
  <input
    checked={false}
    onClick={[Function]}
    type="checkbox"
  />
  <span>
    title
  </span>
</div>
`;
